pkg/asset/permissions/permissions.go

```diff
@@ -44,35 +47,52 @@ func (o *Permissions) Generate(ctx context.Context, dependencies asset.Parents)
 	switch platform {
 	case aws.Name:
 		reqGroups := awsconfig.RequiredPermissionGroups(ic.Config)
-		perms, err := awsconfig.PermissionsList(reqGroups)
-		if err != nil {
-			return fmt.Errorf("failed to generate permissions list: %w", err)
-		}
-		policy := iamv1.PolicyDocument{
-			Version: "2012-10-17",
-			Statement: []iamv1.StatementEntry{
-				{
-					Effect:   "Allow",
-					Action:   perms,
-					Resource: iamv1.Resources{"*"},
-				},
-			},
+		// Include permissions needed by CCO/cluster for mint creds mode
+		mintGroups := append([]awsconfig.PermissionGroup{awsconfig.PermissionMintCreds}, reqGroups...)
+		if err := o.writePolicy(mintGroups, awsPolicyMintFilename); err != nil {
+			return fmt.Errorf("failed to generate mint credentials permissions: %w", err)
 		}
-		policyBytes, err := json.Marshal(policy)
-		if err != nil {
-			return fmt.Errorf("failed to marshal permissions policy: %w", err)
+		// Include permissions needed by CCO/cluster for passthrough creds mode
+		passGroups := append([]awsconfig.PermissionGroup{awsconfig.PermissionPassthroughCreds}, reqGroups...)
+		if err := o.writePolicy(passGroups, awsPolicyPassthroughFilename); err != nil {
+			return fmt.Errorf("failed to generate passthrough credentials permissions: %w", err)
 		}
```

Comment:
we know from the install config which mode the CCO will run in, so should we just create the policy for the specified mode, rather than both?

Reply:
Interesting. If we know, then why do we do [this](https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/aws/permissions.go#L332-L351)?

Reply:
Would it be to cover the `ManualCredentials` case?

Reply:
Fixed.